### ISSUE TITLE
First menu item is not spoken when pressing application key

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#8071]
### Steps to reproduce:
1. Open Outlook and ensure that you land on a list of messages
2. Press application key on your keyboard to open the context menu for a message

### Expected behavior:
NVDA should announce the first menu item 

### Actual behavior:
In Outlook 2007, NVDA is totally silent, in Outlook 2016 NVDA reports "Window, Menu" but not the menu entry. I have to press the arrow keys to get the information.

### System configuration:
NVDA version:
2018.1

NVDA Installed or portable:
Both

Other information:

Windows version:
Windows 10.16299.248, ^

Name and version of other software in use when reproducing the issue:
Outlook 2007, outlook 2016.

### Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
No.

**Comment by Adriani90** [comment]
this seems to be valid for several context menus accross windows. Not only in MS Outlook.

**Comment by Adriani90** [comment]
The only case where it works properly is i.e. when opening a mesage in Outlook or a document in Word and pressing Application key on a letter. Then the focused menu item is reported instantly.

**Comment by michaelDCurran** [comment]
For most menus in Windows, this is by design. When the context menu is 
opened, nothing is actually focused at all.
It would be good to get sighted confirmation that in Outlook, the 
context menu does or doesn't have its first item focused before pressing 
an arrow key. though, I very much doubt something is focused as the 
focus state is not on any of the items.

**Comment by Adriani90** [comment]
Hello,

 

in Jaws for example, I think the first menu entry is spoken when pressing application key. For example on a desktop element “open” on an outlook message “fast print”.

 

 

Von: Michael Curran <notifications@github.com> 
Gesendet: Sonntag, 11. März 2018 22:40
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] The current focused menu item is not spoken when pressing application key (#8071)

 

For most menus in Windows, this is by design. When the context menu is 
opened, nothing is actually focused at all.
It would be good to get sighted confirmation that in Outlook, the 
context menu does or doesn't have its first item focused before pressing 
an arrow key. though, I very much doubt something is focused as the 
focus state is not on any of the items.

**Comment by josephsl** [comment]
Hi, what we’re really after is more accurate announcement. If the first item in a context menu isn’t selected, then NVDA is correct in that I wouldn’t announce the first item. However, we’re looking for counterexamples where fist item in context menus are selected by Windows, not by screen readers. Thanks.

**Comment by Adriani90** [comment]
Well in this case NVDA should not announce the first item in the explorer when pressing win+e. Because you have to press an arrow key to focus an item.

 

 

Von: Joseph Lee <notifications@github.com> 
Gesendet: Dienstag, 13. März 2018 19:39
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] The current focused menu item is not spoken when pressing application key (#8071)

 

Hi, what we’re really after is more accurate announcement. If the first item in a context menu isn’t selected, then NVDA is correct in that I wouldn’t announce the first item. However, we’re looking for counterexamples where fist item in context menus are selected by Windows, not by screen readers. Thanks.

**Comment by Adriani90** [comment]
I have edited the issue accordingly.

**Comment by Adriani90** [comment]
Closing as duplicate of #4231
